src/Microsoft.Identity.Web/DownstreamWebApiSupport/MicrosoftIdentityAuthenticationDelegatingHandlerFactory.cs

```diff
@@ -0,0 +1,73 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Net.Http;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Options;
+
+namespace Microsoft.Identity.Web
+{
+    /// <summary>
+    /// Interface to a class that provides the <see cref="DelegatingHandler"/> that
+    /// addes an authorization header with a token for the application.
+    /// </summary>
+    public interface IMicrosoftIdentityAuthenticationDelegatingHandlerFactory
+    {
+        /// <summary>
+        /// Creates an instance of a <see cref="DelegatingHandler"/> that adds
+        /// an authorization header with a token for an application.
+        /// </summary>
+        /// <param name="serviceProvider">
+        /// The <see cref="IServiceProvider"/> to resolve dependencies from.
+        /// </param>
+        /// <param name="serviceName">
+        /// Name of the service describing the downstream web API. Used to
+        /// retrieve the appropriate config section.
+        /// </param>
+        /// <returns>
+        /// The <see cref="DelegatingHandler"/>.
+        /// </returns>
+        DelegatingHandler CreateAppHandler(IServiceProvider serviceProvider, string? serviceName);
```

Comment:
The `IServiceProvider` should not be given as a parameter of the method but managed by the class implementing the interface (maybe some implementation doesn't need an IServiceProvider)

Reply:
Good idea! I removed `IServiceProvider` from the interface's signature and added it for constructor injection in the default implementation.